src/sparselm/model/_base.py

```diff
@@ -101,47 +126,111 @@ def fit(
             instance of self
         """
         X, y = self._validate_data(
-            X, y, accept_sparse=False, y_numeric=True, multi_output=True
+            X, y, accept_sparse=False, y_numeric=True, multi_output=False
         )
 
         if sample_weight is not None:
             sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+            # rescale sample_weight to sum to number of samples
+            sample_weight = sample_weight * (X.shape[0] / np.sum(sample_weight))
 
-        X, y, X_offset, y_offset, X_scale = self._preprocess_data(
-            X, y, copy=self.copy_X, sample_weight=sample_weight
+        X, y, X_offset, y_offset, X_scale = _preprocess_data(
+            X,
+            y,
+            copy=self.copy_X,
+            fit_intercept=self.fit_intercept,
+            sample_weight=sample_weight,
         )
 
         if sample_weight is not None:
             X, y, _ = _rescale_data(X, y, sample_weight)
 
-        self.coef_ = self._solve(X, y, *args, **kwargs)
+        self._validate_params(X, y)
```

Comment:
BaseEstimator _validate_params does not take into account X or y. Are you sure it is absolutely necessary to break the fingerprint?

Reply:
Yes, for example, all estimators that have a group parameter must have axes to X in order to correctly validate. 
I could rename the method to something else so that it is not redefining `BaseEstimator._validate_params`, but essentially it serves the same purpose.

Reply:
Okay. Then maybe I should implement the same in stepwise.py as well.